### PR TITLE
Add unit tests for com.jme3.renderer.Camera

### PR DESCRIPTION
**Pull request by sudo-jaa**
Also added additional` .equals()` methods for Camera.java and Plane.java to assist testing of the Camera class. 

This PR contains a suite of tests for `com.jme3.renderer.Camera`, once again utilising [Diffblue](https://www.diffblue.com/) [Cover](https://www.diffblue.com/datasheet) to assist with the testing workflow. The rest of the requested classes and packages are on my radar; i'll be looking into them next.

### CODE REVIEW

File: jme3-core/src/main/java/com/jme3/math/Plane.java [hunk 1]
```diff
@@ -291,4 +292,13 @@ public Plane clone() {
             throw new AssertionError();
         }
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Plane plane = (Plane) o;
+        return Float.compare(plane.constant, constant) == 0 &&
```
Comment (freddytuxworth) [inline jme3-core/src/main/java/com/jme3/math/Plane.java]: replace `Float.compare(...)` with `plane.constant == constant`

File: jme3-core/src/main/java/com/jme3/math/Plane.java [hunk 2]
```diff
@@ -33,6 +33,7 @@
 
 import com.jme3.export.*;
 import java.io.IOException;
+import java.util.Objects;
```
Comment (freddytuxworth) [inline jme3-core/src/main/java/com/jme3/math/Plane.java]: Is this used?

File: jme3-core/src/main/java/com/jme3/renderer/Camera.java [hunk 3]
```diff
@@ -999,6 +1001,7 @@ public float distanceToNearPlane(Vector3f pos) {
         return worldPlane[NEAR_PLANE].pseudoDistance(pos);
     }
 
+
```
Comment (freddytuxworth) [inline jme3-core/src/main/java/com/jme3/renderer/Camera.java]: remove added newline

File: jme3-core/src/test/java/com/jme3/renderer/CameraTest.java [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+        final Camera copyToCamera = new Camera(2, 2);
+
+        baseCamera.setLocation(new Vector3f(1, 2, 3));
+        baseCamera.setFrustum(1, 2, 3, 4, 5, 6);
+        baseCamera.setLocation(new Vector3f(1, 5, 8));
+        baseCamera.setRotation(new Quaternion(8, 9, 10, 11));
+        baseCamera.setViewPort(1, 2, 3, 4);
+        baseCamera.setFrustumPerspective(1.0f, 2.5f, 6.5f, 8.5f);
+
+        assertNotEquals(baseCamera, copyToCamera);
```
Comment (freddytuxworth) [inline jme3-core/src/test/java/com/jme3/renderer/CameraTest.java]: Why?

File: jme3-core/src/test/java/com/jme3/renderer/CameraTest.java [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+
+        camera.resize(2, 3, false);
+
+        assertEquals(1.0, camera.frustumNear, 0.0f);
+        assertEquals(2.0, camera.frustumFar, 0.0f);
+        assertEquals(-0.5, camera.frustumLeft, 0.0f);
+        assertEquals(0.5, camera.frustumRight, 0.0f);
+        assertEquals(0.5, camera.frustumTop, 0.0f);
+        assertEquals(-0.5, camera.frustumBottom, 0.0f);
+
```
Comment (freddytuxworth) [inline jme3-core/src/test/java/com/jme3/renderer/CameraTest.java:58]: replace all `.frustum` with `.getFrustum`

File: jme3-core/src/test/java/com/jme3/renderer/CameraTest.java [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+        assertEquals(camera.getWidth(), 2);
+        assertEquals(camera.getHeight(), 3);
+
+        camera.resize(2, 3, true);
```
Comment (freddytuxworth) [inline jme3-core/src/test/java/com/jme3/renderer/CameraTest.java]: If you use `resize(2, 4, true)`, do the `0.333`s become `0.5`s and can you drop the `delta` to 0?

File: jme3-core/src/test/java/com/jme3/renderer/CameraTest.java [hunk 7]
```diff
@@ -0,0 +1,276 @@
...
+        assertEquals(camera.getHeight(), 3);
+    }
+
+    @Test
+    public void testLookAt() {
+        final Vector3f pos = new Vector3f(1, 2, 3);
+        final Vector3f worldUpVector = new Vector3f(30, 40, 50);
+        final Camera camera = new Camera(1, 1);
+
+        assertEquals(new Quaternion(0.0f, 0.0f, 0.0f, 1.0f), camera.getRotation());
```
Comment (freddytuxworth) [inline jme3-core/src/test/java/com/jme3/renderer/CameraTest.java]: Remove assertion about rotation before `lookAt` is called, because the default might change

File: jme3-core/src/test/java/com/jme3/renderer/CameraTest.java [hunk 8]
```diff
@@ -0,0 +1,276 @@
...
+        assertEquals(new Quaternion(-0.31154725f, -0.04523793f, -0.5420604f, 0.77914214f), camera.getRotation());
+    }
+
+    @Test
+    public void testLookAt2() {
+        final Vector3f pos = new Vector3f(1, 2, 3);
+        final Vector3f worldUpVector = new Vector3f(0, 0, 0);
+        final Camera camera = new Camera(1, 1);
+
+        assertEquals(new Quaternion(0.0f, 0.0f, 0.0f, 1.0f), camera.getRotation());
```
Comment (freddytuxworth) [inline jme3-core/src/test/java/com/jme3/renderer/CameraTest.java]: Remove assertion about rotation before `lookAt` is called, because the default might change

File: jme3-core/src/test/java/com/jme3/renderer/CameraTest.java [hunk 9]
```diff
@@ -0,0 +1,276 @@
...
+    public void testContainsGui() {
+        final Camera camera = new Camera(1, 5);
+
+        assertTrue(camera.containsGui(null));
+
+        final BoundingVolume bounds1 = new BoundingBox(new Vector3f(1.0f, 2.0f, 3.0f), 4.0f, 9.0f, 4.5f);
+        assertTrue(camera.containsGui(bounds1));
+
+        final BoundingVolume bounds2 = new BoundingSphere(-50.0f, new Vector3f(20.0f, 10.0f, 3.0f));
+        assertTrue(camera.containsGui(bounds2));
```
Comment (freddytuxworth) [inline jme3-core/src/test/java/com/jme3/renderer/CameraTest.java:193]: a test returning false would be good

**Review by freddytuxworth — APPROVED** [review]

File: jme3-core/src/test/java/com/jme3/renderer/CameraTest.java (hunk 5 above)
Comment (sudo-jaa) [reply jme3-core/src/test/java/com/jme3/renderer/CameraTest.java:58]: Repo owner prefer using direct field access over using the getters. I'm not sure why, exactly.

File: jme3-core/src/main/java/com/jme3/math/Plane.java [hunk 10]
```diff
@@ -291,4 +291,13 @@ public Plane clone() {
             throw new AssertionError();
         }
     }
+
+    @Override
```
Comment (peterschrammel) [inline jme3-core/src/main/java/com/jme3/math/Plane.java:295]: Put changes to this file into a separate commit to separate from the tests

File: jme3-core/src/main/java/com/jme3/renderer/Camera.java [hunk 11]
```diff
@@ -1480,4 +1482,38 @@ public void read(JmeImporter e) throws IOException {
         onViewPortChange();
         onFrameChange();
     }
+
```
Comment (peterschrammel) [inline jme3-core/src/main/java/com/jme3/renderer/Camera.java:1485]: Put changes to this file into a separate commit to separate from the tests

**Review by freddytuxworth — APPROVED** [review]

**Review by petersear — APPROVED** [review]
